File: binding-library/java/src/main/java/com/microsoft/azure/functions/kafka/BrokerAuthenticationMode.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */
package com.microsoft.azure.functions.kafka;

/**
 * Defines the broker authentication modes
 */
public enum BrokerAuthenticationMode {
    NOTSET(-1),
    GSSAPI(0),
    PLAIN(1),
    SCRAMSHA256(2),
    SCRAMSHA512(3);

    private final int value;

    BrokerAuthenticationMode(final int value) {
        this.value = value;
    }

    public int getValue() { return value; }
}

File: src/Microsoft.Azure.WebJobs.Extensions.Kafka/Config/BrokerAuthenticationMode.cs
﻿// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System;
using Confluent.Kafka;

namespace Microsoft.Azure.WebJobs.Extensions.Kafka
{
    /// <summary>
    /// Defines the broker authentication modes
    /// </summary>
    public enum BrokerAuthenticationMode
    {
        // Force that 0 starts like the one from librdkafka
        NotSet = -1,
        Gssapi,
        Plain,
        ScramSha256,
        ScramSha512,
        OAuthBearer
    }
}
